ExotischNederland/Menus/RouteMenu.cs

```diff
@@ -0,0 +1,319 @@
+using System;
+using System.Collections.Generic;
+using ExotischNederland.Models;
+
+namespace ExotischNederland.Menus
+{
+    internal class RouteMenu
+    {
+        private readonly User authenticatedUser;
+        private Dictionary<string, string> menuItems = new Dictionary<string, string>();
+
+        public RouteMenu(User _authenticatedUser)
+        {
+            this.authenticatedUser = _authenticatedUser;
+        }
+
+        public Dictionary<string, string> GetMenuItems()
+        {
+            Dictionary<string, string> menuItems = new Dictionary<string, string>();
+            
+            var permission = authenticatedUser.Permission;
+
+            if (permission.CanViewRoutes())
+                menuItems.Add("viewAllRoutes", "View All Routes");
+            if (permission.CanCreateRoute())
+                menuItems.Add("createRoute", "Create Route");
+            if (permission.CanEditRoute(null))
+                menuItems.Add("editRoute", "Edit Route");
+            if (permission.CanDeleteRoute(null))
+                menuItems.Add("deleteRoute", "Delete Route");
+            if (permission.CanEditRoute(null))
+                menuItems.Add("manageRoutePoints", "Manage Route Points");
+
+            menuItems.Add("back", "Return to main menu");
+            return menuItems;
+        }
+
+        public void Show()
+        {
+            menuItems = this.GetMenuItems();
+            while (true)
+            {
+                string selected = Helpers.MenuSelect(this.menuItems, true);
+
+                if (selected == "viewAllRoutes")
+                {
+                    ViewAllRoutes();
+                }
+                else if (selected == "createRoute" && authenticatedUser.Permission.CanCreateRoute())
```

Comment:
The extra authorization check is not necessary here. The menu item will not show up without the permission, the use can only select items, not input their own string and the action of creating a route should be protected at the model level anyways